webapp/src/store/Store.ts

```diff
@@ -1,7 +1,82 @@
+import fs from 'fs/promises';
+
 import { ProjectStore } from '@/store/ProjectStore';
+import MessageAdapterFactory from '@/utils/adapters/MessageAdapterFactory';
+import YamlTranslationAdapter from '@/utils/adapters/YamlTranslationAdapter';
+import { LyraProjectConfig } from '@/utils/lyraConfig';
+import { StoreData } from './types';
+
+const FILE_PATH = './store.json';
 
 export class Store {
   private data = new Map<string, ProjectStore>();
+  private initialState: Record<string, StoreData | undefined> = {};
+  private writes: Promise<void> = Promise.resolve();
+
+  public static async getProjectStore(
+    lyraProjectConfig: LyraProjectConfig,
+  ): Promise<ProjectStore> {
+    /**
+     * As this function is not async,
+     * no concurrent task executes between checking and updating globalThis.
+     */
+    function initialize(): Promise<Store> {
+      if (!globalThis.store) {
+        const newStore = new Store();
+        globalThis.store = newStore
+          .loadFromDisk()
+          .catch((reason) => {
+            // Forget the promise, so that the next call will retry.
+            globalThis.store = null;
+            throw reason;
+          })
+          .then(() => newStore);
+      }
+      return globalThis.store;
+    }
+    const store = await initialize();
+
+    if (!store.hasProjectStore(lyraProjectConfig.absPath)) {
+      const initialProjectState = store.initialState[lyraProjectConfig.absPath];
+
+      const projectStore = new ProjectStore(
+        MessageAdapterFactory.createAdapter(lyraProjectConfig),
+        new YamlTranslationAdapter(lyraProjectConfig.absTranslationsPath),
+        initialProjectState,
+      );
+      store.addProjectStore(lyraProjectConfig.absPath, projectStore);
+    }
+
+    return store.getProjectStore(lyraProjectConfig.absPath);
+  }
+
+  public static async persistToDisk(): Promise<void> {
+    const store = await globalThis.store;
+    if (!store) {
+      return;
+    }
+
+    // Retain initial data if we have not opened a ProjectStore
+    const payload = {};
+    Object.assign(payload, store.initialState, store.toJSON());
```

Comment:
With this solution, we will never forget projects, even if they have been removed from `projects.yaml`

We will open a separate issue for that.